### PR TITLE
Smart ghost finished

### PR DESCRIPTION
**Pull request by ShalomPrinz**
not the best, but for now it's ready

### CODE REVIEW

File: DeadGhost.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import pacman.Game.Direction;
+
+public class DeadGhost extends Ghost {
+	
+	public void findPath(Game game, Location goal) {
+				
+		for (Direction d : Direction.values()) {
+			Location nextLocation = game.getNextLocation(getLocation(), d);
+			if ( nextLocation.equals(goal) )
```
Comment (danprinz) [inline DeadGhost.java:11]: this means that only if the ghost is one step next to the goal it will set the direction, isn't it?
you tested it only with one path available - "WD-R"
what about:
 "W----W"
 "R--D--"
 "W----W"

File: SmartGhost.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+		if (nextCreature.getType() == Type.PACMAN)
+			game.ghostMeetPacman( (Pacman) nextCreature, this, false);
+		else
+			game.set(this);	
+	}
+	
+	public void changeDirection(Game game) {
+
+		Direction forbiddenDirection = getOppositeDirection( getDirection() ), check;
```
Comment (danprinz) [inline SmartGhost.java]: don't mix variable declarations.
even if syntax is correct, it's a bad practice.

File: SmartGhost.java [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+		
+		setSpeeds( getDirection() );
+		
+		if (horizonalSpeed == 0) {
```
Comment (danprinz) [inline SmartGhost.java]: while it's very hard to follow your logic, I absolutely sure that you can write the very **same logic** more readable. 
and your logic is complex, please add some nice comment above to explain it

**Review by danprinz — CHANGES_REQUESTED** [review]

File: DeadGhost.java (hunk 1 above)
Comment (ShalomPrinz) [reply DeadGhost.java:11]: 5th test in DeadGhostTest: chooseTheOneStep - confirms one step path choosing, for all directions.
